### PR TITLE
Week 9-12 Progress Update: Tan Heng Yeow

### PR DESCRIPTION
**Pull request by tanhengyeow**
Hi @wkurniawan07 @tran-tien-dat

Updated my progress for Week 9-12, kindly review :)

### CODE REVIEW

**Review by wkurniawan07 — APPROVED** [review]
LGTM, @tran-tien-dat please merge after your approval

File: Tan_Heng_Yeow/progress.md
Comment (tran-tien-dat) [inline Tan_Heng_Yeow/progress.md:214]: I don't see how introducing HTML poses a risk of XSS. XSS is a risk when we display user content, regardless of whether it is HTML or not. If the HTML added is totally controlled by our app (e.g. error messages without repeating user content) then that would not be a XSS risk. so which case is yours?

File: Tan_Heng_Yeow/progress.md
Comment (tran-tien-dat) [inline Tan_Heng_Yeow/progress.md]: this is not called "refactoring". "Refactor" means you do not change the external behaviour, just the internal mechanism. In this case, you are changing the external behaviour, isn't it?

**Review by tran-tien-dat — CHANGES_REQUESTED** [review]
Some comments

File: Tan_Heng_Yeow/progress.md
Comment (tanhengyeow) [reply Tan_Heng_Yeow/progress.md:214]: This was mentioned during the time I introduced status icons in the spreadsheet through the use of Handontable `HTMLRenderer` which allows any HTML content to be displayed in the spreadsheet (for that row/column which uses it). An user can enter `<script>` or other potentially malicious tags using the cell editor in these fields.

The point made refers to the possibility of user sending crafted HTML content to the backend and when left unsanitized would lead to a XSS attack. 

This is not relevant now since I am not using the `HTMLRenderer` anymore but instead using `TextRenderer` for the error messages.

File: Tan_Heng_Yeow/progress.md
Comment (tanhengyeow) [reply Tan_Heng_Yeow/progress.md]: Yup you're right, I shall change "refactored" to "modified" 😅

**Comment by tran-tien-dat** [comment]
I see, so the old approach was to inject HTML into the cell content.

**Review by tran-tien-dat — APPROVED** [review]

**Pull request merged**
